src/org/labkey/targetedms/folderImport/PanoramaQCSettings.java

```diff
@@ -0,0 +1,489 @@
+                                        properties.put(entry.getKey().toString(), String.valueOf(metricRowId));
+                                    }
+                                    else
+                                    {
+                                        properties.put(entry.getKey().toString(), entry.getValue().toString());
+                                    }
+                                }
+                            }
+                            properties.save();
+                            numProps = props.size();
+                        }
+                    }
+                    return numProps;
+                }
+            },
+    GUIDE_SET (TargetedMSSchema.TABLE_GUIDE_SET, QCFolderConstants.GUIDE_SET_FILE_NAME),
+    PRECURSOR_EXCLUSION (TargetedMSSchema.TABLE_PEPTIDE_MOLECULE_PRECURSOR_EXCLUSION, QCFolderConstants.PEPTIDE_MOLECULE_PRECURSOR_EXCLUSION_FILE_NAME);
+
+    private final @Nullable String _tableName;
+    private final String _settingsFileName;
+    private static final Logger LOG = LogHelper.getLogger(PanoramaQCSettings.class, "Panorama QC Folder Settings");
+
+    PanoramaQCSettings(@Nullable String tableName, String fileName)
+    {
+        _tableName = tableName;
+        _settingsFileName = fileName;
+    }
+
+    public @Nullable String getTableName()
+    {
+        return _tableName;
+    }
+
+    public String getSettingsFileName()
+    {
+        return _settingsFileName;
+    }
+
+    public void writeSettings(VirtualFile vf, Container container, User user) throws Exception
+    {
+        TargetedMSSchema schema = new TargetedMSSchema(user, container);
+        TableInfo ti = schema.getTable(getTableName());
+        assert ti != null;
+        List<ColumnInfo> userEditableCols = ti.getColumns().stream().filter(ci -> ci.isUserEditable()).collect(Collectors.toList());
+        ResultsFactory factory = ()-> QueryService.get().select(ti, userEditableCols, null, null);
+        writeSettingsToTSV(vf, factory, getSettingsFileName(), getTableName());
+    }
+
+    protected void writeSettingsToTSV(VirtualFile vf, ResultsFactory factory, String fileName, String tableName) throws Exception
+    {
+        try
+        {
+            if (factory.get().countAll() > 0)
+            {
+                try (TSVGridWriter tsvWriter = new TSVGridWriter(factory))
+                {
+                    tsvWriter.setApplyFormats(false);
+                    tsvWriter.setColumnHeaderType(ColumnHeaderType.FieldKey);
+                    PrintWriter out = vf.getPrintWriter(fileName);
+                    tsvWriter.write(out);
+                }
+                catch (IOException e)
+                {
+                    throw new IOException("Error writing results to " + fileName, e);
```

Comment:
Can this be consolidated? Seems overkill to catch IOException, throw a new IOException, and then catch that all within 5 lines